File: jQuery-deprecated/public/js-backend/movement/MovementFactory.js
const MovementType = require("./MovementType").MovementType;
const Move = require("./Move");
const Jump = require("./Jump");
const Slide = require("./Slide");
const JumpSlide = require("./JumpSlide");
const Strike = require("./Strike");
const Command = require("./Command");
const Summon = require("./Summon");

class MovementFactory {
	constructor() {
		this.move = new Move();
		this.jump = new Jump();
		this.slide = new Slide();
		this.jumpSlide = new JumpSlide();
		this.strike = new Strike();
		this.command = new Command();
		this.summon = new Summon();
	}

	createMovement(type) {
		switch (type)
		{
		case MovementType.MOVE:
			return this.move;
		case MovementType.JUMP:
			 return this.jump;
		case MovementType.SLIDE:
			return this.slide;
		case MovementType.JUMPSLIDE:
			return this.jumpSlide;
		case MovementType.STRIKE:
			return this.strike;
		case MovementType.COMMAND:
			return this.command;
		case MovementType.SUMMON:
			return this.summon;
		default: 
			return null;
		}
	}
}

module.exports = MovementFactory;

File: deprecated/Java/com.dexaint.dukechess/src/main/java/com/dexaint/dukechess/movement/MovementFactory.java
package com.dexaint.dukechess.movement;

public class MovementFactory {
	private static Move move = new Move();
	private static Jump jump = new Jump();
	private static Slide slide = new Slide();
	private static JumpSlide jumpSlide = new JumpSlide();
	private static Strike strike = new Strike();
	private static Command command = new Command();
	private static Summon summon = new Summon();
	
	public static Movement createMovement(MovementType type) {
		switch (type)
		{
		case Move:
			return move;
		case Jump:
			 return jump;
		case Slide:
			return slide;
		case JumpSlide:
			return jumpSlide;
		case Strike:
			return strike;
		case Command:
			return command;
		case Summon:
			return summon;
		default:
			return null;
		}
	}
}
